src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,233 @@
+	for ; numberTimes > 0; numberTimes-- {
+		fullPath = path.Dir(fullPath)
+	}
+	return fullPath
+}
+
+func getWindowsRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	localAppDataGetters := []func() string{
+		func() string {
+			return os.Getenv("LOCALAPPDATA")
+		},
+		func() string {
+			homeDrive := os.Getenv("HOMEDRIVE")
+			if homeDrive == "" {
+				return ""
+			}
+			homePathPart := os.Getenv("HOMEPATH")
+			if homePathPart == "" {
+				return ""
+			}
+			return path.Join(homeDrive+homePathPart, "AppData", "Local")
+		},
+		func() string {
+			homePath := os.Getenv("HOME")
+			if homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+		func() string {
+			homePath, err := os.UserHomeDir()
+			if err != nil || homePath == "" {
+				return ""
+			}
+			return path.Join(homePath, "AppData", "Local")
+		},
+	}
+	for _, localAppDataGetter := range localAppDataGetters {
+		localAppDataDir := localAppDataGetter()
+		candidatePath := checkExistence(path.Join(localAppDataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return ""
+}
+
+func getMacOSRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		// we're at .../Applications/R D.app/Contents/Resources/resources/darwin/bin
+		// and want to move to /Applications (or ~/Applications, or something else)
+		normalParentPath := moveToParent(rdctlPath, 7)
+		candidatePath := checkExistence(path.Join(normalParentPath, "Rancher Desktop.app"), 0)
+		if candidatePath != "" {
+			// No need to verify permissions on the directory App/R D.app -- it must be a directory, or it
+			// couldn't be the ancestor parent of rdctl. And it must be executable/readable enough for us
+			// to have been able to path.Dir() our way up to it.
+			return candidatePath
+		}
+	}
+	return checkExistence(path.Join("/Applications", "Rancher Desktop.app"), 0)
+}
+
+func getLinuxRDPath(rdctlPath string) string {
+	if rdctlPath != "" {
+		normalParentPath := moveToParent(rdctlPath, 5)
+		candidatePath := checkExistence(path.Join(normalParentPath, "rancher-desktop"), 0o111)
+		if candidatePath != "" {
+			return candidatePath
+		}
+	}
+	return checkExistence("/opt/rancher-desktop/rancher-desktop", 0111)
+}
+
+/**
+ * Verify the path exists. For Linux pass in mode bits to guarantee the file is executable (for at least one
+ * category of user). Note that on macOS the candidate is a directory, so never pass in mode bits.
+ * And mode bits don't make sense on Windows.
+ */
+func checkExistence(candidatePath string, modeBits fs.FileMode) string {
+	stat, err := os.Stat(candidatePath)
+	if err != nil {
+		return ""
+	}
+	if modeBits != 0 && (!stat.Mode().IsRegular() || stat.Mode().Perm()&modeBits == 0) {
```

Comment:
Should that be `…&modeBits == 0` or `…&modeBits != modeBits`?
That is, if the caller asks for `0o755` and the file is actually `0o644`, should it be accepted?

(I think this just needs to be documented better.)

Reply:
I'm only using this to check for executability. If the file isn't readable someone's done something beyond our control.